### PR TITLE
SQLite logging for investigating row-level locking.

### PR DESCRIPTION
**Pull request by tylerkaraszewski**
Implements new logging added in sqlite.

Example logs, collected by adding `-pageLogging 1` to `auth.service` in dev, and to `BedrockTester`s default args.

```
2020-03-31T21:05:11.228950+00:00 expensidev bedrock: PWJkD4 (SQLite.cpp:783) commit [worker5] [row-level-locking] transaction attempt:71 committed. report: W:{8 11 12 14}#012R:{8 11 12 14}#0128<-(73){NULL,'DELETE FROM jobs WHERE jobID > 0;','32ADBF3F90E9C18EB946C5F55AB54EDA3E12C6E7'}#0128:[73..73]#01211<-(4805827903235208815){'2020-03-31 21:05:11',NULL,'RUNNING','high_2','2020-03-31 21:05:11','2020-03-31 21:05:11'}#01211<-(4187467783712293097){'2020-03-31 21:05:11',NULL,'RUNNING','high_1','2020-03-31 21:05:11','2020-03-31 21:05:11'}#01211<-(1829565733273234435){'2020-03-31 21:05:11',NULL,'RUNNING','medium_3','2020-03-31 21:05:11','2020-03-31 21:05:11'}#01211<-(1630222742581996923){'2020-03-31 21:05:11',NULL,'RUNNING','low_5','2020-03-31 21:05:11','2020-03-31 21:05:11'}#01211<-(1119442079297440770){'2020-03-31 21:05:11',NULL,'RUNNING','medium_4','2020-03-31 21:05:11','2020-03-31 21:05:11'}#01211:(0..EOF)
2020-03-31T21:05:11.233585+00:00 expensidev bedrock: 32LLfD (SQLite.cpp:783) commit [worker7] [row-level-locking] transaction attempt:72 committed. report: W:{10 11 12 14}#012R:{10 11 12 14}#01210<-(74){NULL,'INSERT INTO jobs ( jobID, created, state, name, nextRun, repeat, data, priority, parentJobID, retryAfter ) VALUES( 3588877378925522013, ''2020-03-31 21:05:11'', ''QUEUED'', ''ParentJob'', ''2020-03-31 21:05:11'', '''', ''{}'', 500, 0, ''+2 SECONDS''  );','9496AF2F0C4EBB0C63A96D7AB131887F52BEF7CA'}#01210:[74..74]#01211<-(3588877378925522013){'2020-03-31 21:05:11',NULL,'QUEUED','ParentJob','2020-03-31 21:05:11',NULL}#01211:[3588877378925522013..3588877378925522013]#01211:[3588877378925522013..3588877378925522013]
2020-03-31T21:05:11.236564+00:00 expensidev bedrock: urgSPu (SQLite.cpp:783) commit [worker1] [row-level-locking] transaction attempt:73 committed. report: W:{4 11 14}#012R:{4 11 13 14}#0124<-(75){NULL,'UPDATE jobs SET state=''RUNQUEUED'', lastRun=''2020-03-31 21:05:11'', nextRun=DATETIME(''2020-03-31 21:05:11'', ''+2 SECONDS'') WHERE jobID = ''3588877378925522013'';','00C4478C73D5F18C08B08449A8EEE2723DA1B67D'}#0124:[75..75]#01211<-(3588877378925522013){'2020-03-31 21:05:11',NULL,'RUNQUEUED','ParentJob','2020-03-31 21:05:13','2020-03-31 21:05:11'}#01211<-(3588877378925522013){'2020-03-31 21:05:11',NULL,'QUEUED','ParentJob','2020-03-31 21:05:11',NULL}#01211:[3588877378925522013..3588877378925522013]#01213:[{3588877378925522013,'CANCELLED'}..{3588877378925522013,'CANCELLED'}]#01214:[{'RUNQUEUED',0}..EOF)#01214:[{'QUEUED',0}..{'QUEUED',0,'2020-03-31 21:05:11'}]#01214:[{'RUNQUEUED',500}..EOF)#01214:[{'QUEUED',500}..{'QUEUED',500,'2020-03-31 21:05:11'}]#01214:[{'RUNQUEUED',1000}..EOF)#01214:[{'QUEUED',1000}..EOF)
2020-03-31T21:05:11.240708+00:00 expensidev bedrock: 12UhUd (SQLite.cpp:783) commit [worker6] [row-level-locking] transaction attempt:74 committed. report: W:{9 11 12 13 14}#012R:{9 11 12 13 14}#0129<-(76){NULL,'INSERT INTO jobs ( jobID, created, state, name, nextRun, repeat, data, priority, parentJobID, retryAfter ) VALUES( 7817508156870089629, ''2020-03-31 21:05:11'', ''PAUSED'', ''ChildJob'', ''2020-03-31 21:05:11'', '''', ''{}'', 500, 3588877378925522013, ''''  );','4863FC9625C6BEA557AB55F6CEB09773DD4AD980'}#0129:[76..76]#01211<-(7817508156870089629){'2020-03-31 21:05:11',NULL,'PAUSED','ChildJob','2020-03-31 21:05:11',NULL}#01211:[7817508156870089629..7817508156870089629]#01211:[7817508156870089629..7817508156870089629]#01211:[3588877378925522013..3588877378925522013]#01211:[3588877378925522013..3588877378925522013]#01211:[3588877378925522013..3588877378925522013]
2020-03-31T21:05:14.247670+00:00 expensidev bedrock: rZldAp (SQLite.cpp:783) commit [worker7] [row-level-locking] transaction attempt:75 committed. report: W:{10 11 14}#012R:{10 11 13 14}#01210<-(77){NULL,'UPDATE jobs SET state=''RUNQUEUED'', lastRun=''2020-03-31 21:05:14'', nextRun=DATETIME(''2020-03-31 21:05:14'', ''+2 SECONDS'') WHERE jobID = ''3588877378925522013'';','D3EA113F9D95FF3B815730F4E5B20EA7075FC36C'}#01210:[77..77]#01211<-(3588877378925522013){'2020-03-31 21:05:11',NULL,'RUNQUEUED','ParentJob','2020-03-31 21:05:16','2020-03-31 21:05:14'}#01211<-(3588877378925522013){'2020-03-31 21:05:11',NULL,'RUNQUEUED','ParentJob','2020-03-31 21:05:13','2020-03-31 21:05:11'}#01211:[3588877378925522013..3588877378925522013]#01213:[{3588877378925522013,'FINISHED'}..{3588877378925522013,'FINISHED'}]#01213:[{3588877378925522013,'CANCELLED'}..{3588877378925522013,'CANCELLED'}]#01214:[{'RUNQUEUED',0}..{'RUNQUEUED',0,'2020-03-31 21:05:14'}]#01214:[{'QUEUED',0}..{'QUEUED',0,'2020-03-31 21:05:14'}]#01214:[{'RUNQUEUED',500}..{'RUNQUEUED',500,'2020-03-31 21:05:14'}]#01214:[{'QUEUED',500}..{'QUEUED',500,'2020-03-31 21:05:14'}]#01214:[{'RUNQUEUED',1000}..EOF)#01214:[{'QUEUED',1000}..{'QUEUED',1000,'2020-03-31 21:05:14'}]
2020-03-31T21:05:14.256421+00:00 expensidev bedrock: rqC7Yz (SQLite.cpp:783) commit [worker1] [row-level-locking] transaction attempt:76 committed. report: W:{4 11 13 14}#012R:{4 11 13 14}#0124<-(78){NULL,'UPDATE jobs SET state=''PAUSED'', nextRun=''2020-03-31 21:05:14'' WHERE jobID=3588877378925522013;UPDATE jobs SET state=''QUEUED'' WHERE state=''PAUSED'' AND parentJobID != 0 AND parentJobID=3588877378925522013;','E1720259721F8615BE9BDC807941CEBE7308B281'}#0124:[78..78]#01211<-(7817508156870089629){'2020-03-31 21:05:11',NULL,'QUEUED','ChildJob','2020-03-31 21:05:11',NULL}#01211<-(7817508156870089629){'2020-03-31 21:05:11',NULL,'PAUSED','ChildJob','2020-03-31 21:05:11',NULL}#01211:[7817508156870089629..7817508156870089629]#01213:[{3588877378925522013,'PAUSED'}..{3588877378925522013,'PAUSED'}]#01211<-(3588877378925522013){'2020-03-31 21:05:11',NULL,'PAUSED','ParentJob','2020-03-31 21:05:14','2020-03-31 21:05:14'}#01211<-(3588877378925522013){'2020-03-31 21:05:11',NULL,'RUNQUEUED','ParentJob','2020-03-31 21:05:16','2020-03-31 21:05:14'}#01211:[3588877378925522013..3588877378925522013]#01213:[{3588877378925522013,'PAUSED'}..{3588877378925522013,'PAUSED'}]#01213:[{3588877378925522013,'FINISHED'}..{3588877378925522013,'FINISHED'}]#01213:[{3588877378925522013,'CANCELLED'}..{3588877378925522013,'CANCELLED'}]#01211:[3588877378925522013..3588877378925522013]
2020-03-31T21:05:14.259355+00:00 expensidev bedrock: WEGlbX (SQLite.cpp:783) commit [worker3] [row-level-locking] transaction attempt:77 committed. report: W:{6 11 13 14}#012R:{6 11 13 14}#0126<-(79){NULL,'UPDATE jobs SET state=''RUNNING'', lastRun=''2020-03-31 21:05:14'' WHERE jobID IN (''7817508156870089629'');','2666E294965C34E473D0C83C521BCB7989C6D775'}#0126:[79..79]#01211<-(7817508156870089629){'2020-03-31 21:05:11',NULL,'RUNNING','ChildJob','2020-03-31 21:05:11','2020-03-31 21:05:14'}#01211<-(7817508156870089629){'2020-03-31 21:05:11',NULL,'QUEUED','ChildJob','2020-03-31 21:05:11',NULL}#01211:[7817508156870089629..7817508156870089629]#01213:[{7817508156870089629,'CANCELLED'}..{7817508156870089629,'CANCELLED'}]#01211:[3588877378925522013..3588877378925522013]#01214:[{'RUNQUEUED',0}..EOF)#01214:[{'QUEUED',0}..{'QUEUED',0,'2020-03-31 21:05:14'}]#01214:[{'RUNQUEUED',500}..EOF)#01214:[{'QUEUED',500}..{'QUEUED',500,'2020-03-31 21:05:14'}]#01214:[{'RUNQUEUED',1000}..EOF)#01214:[{'QUEUED',1000}..EOF)
2020-03-31T21:05:14.262754+00:00 expensidev bedrock: r2O4jW (SQLite.cpp:783) commit [worker6] [row-level-locking] transaction attempt:78 committed. report: W:{9 11 13 14}#012R:{9 11 13 14}#0129<-(80){NULL,'UPDATE jobs SET state=''FINISHED'' WHERE jobID=7817508156870089629;UPDATE jobs SET state=''QUEUED'' where jobID=3588877378925522013;','B48404D94A8452D1A35A9CF4E9F956552919638B'}#0129:[80..80]#01211<-(3588877378925522013){'2020-03-31 21:05:11',NULL,'QUEUED','ParentJob','2020-03-31 21:05:14','2020-03-31 21:05:14'}#01211<-(3588877378925522013){'2020-03-31 21:05:11',NULL,'PAUSED','ParentJob','2020-03-31 21:05:14','2020-03-31 21:05:14'}#01211:[3588877378925522013..3588877378925522013]#01213:[{3588877378925522013,'RUNQUEUED'}..{3588877378925522013,'RUNQUEUED'}]#01213:[{3588877378925522013,'RUNNING'}..{3588877378925522013,'RUNNING'}]#01213:[{3588877378925522013,'QUEUED'}..{3588877378925522013,'QUEUED'}]#01213:[{3588877378925522013,'PAUSED'}..{3588877378925522013,'PAUSED'}]#01211<-(7817508156870089629){'2020-03-31 21:05:11',NULL,'FINISHED','ChildJob','2020-03-31 21:05:11','2020-03-31 21:05:14'}#01211<-(7817508156870089629){'2020-03-31 21:05:11',NULL,'RUNNING','ChildJob','2020-03-31 21:05:11','2020-03-31 21:05:14'}#01211:[7817508156870089629..7817508156870089629]#01213:[{7817508156870089629,'PAUSED'}..{7817508156870089629,'PAUSED'}]#01213:[{7817508156870089629,'CANCELLED'}..{7817508156870089629,'CANCELLED'}]#01211:[3588877378925522013..3588877378925522013]#01211:[7817508156870089629..7817508156870089629]
2020-03-31T21:05:14.265925+00:00 expensidev bedrock: SnctCN (SQLite.cpp:783) commit [worker4] [row-level-locking] transaction attempt:79 committed. report: W:{7 11 14}#012R:{7 11 13 14}#0127<-(81){NULL,'UPDATE jobs SET state=''RUNQUEUED'', lastRun=''2020-03-31 21:05:14'', nextRun=DATETIME(''2020-03-31 21:05:14'', ''+2 SECONDS'') WHERE jobID = ''3588877378925522013'';','83590AF826EBBA0226236E493382248C5719A0DA'}#0127:[81..81]#01211<-(3588877378925522013){'2020-03-31 21:05:11',NULL,'RUNQUEUED','ParentJob','2020-03-31 21:05:16','2020-03-31 21:05:14'}#01211<-(3588877378925522013){'2020-03-31 21:05:11',NULL,'QUEUED','ParentJob','2020-03-31 21:05:14','2020-03-31 21:05:14'}#01211:[3588877378925522013..3588877378925522013]#01213:[{3588877378925522013,'FINISHED'}..{3588877378925522013,'FINISHED'}]#01213:[{3588877378925522013,'CANCELLED'}..{3588877378925522013,'CANCELLED'}]#01214:[{'RUNQUEUED',0}..EOF)#01214:[{'QUEUED',0}..{'QUEUED',0,'2020-03-31 21:05:14'}]#01214:[{'RUNQUEUED',500}..EOF)#01214:[{'QUEUED',500}..{'QUEUED',500,'2020-03-31 21:05:14'}]#01214:[{'RUNQUEUED',1000}..EOF)#01214:[{'QUEUED',1000}..EOF)
2020-03-31T21:05:14.272129+00:00 expensidev bedrock: C8A7aL (SQLite.cpp:783) commit [worker5] [row-level-locking] transaction attempt:80 committed. report: W:{8 11 12 13 14}#012R:{8 11 12 13 14}#0128<-(82){NULL,'DELETE FROM jobs WHERE parentJobID != 0 AND parentJobID=3588877378925522013 AND state IN (''FINISHED'', ''CANCELLED'');DELETE FROM jobs WHERE jobID=3588877378925522013;','078C3690F203A8989C1DDBD6D2CDD592E14BE242'}#0128:[82..82]#01213:[{3588877378925522013}..{3588877378925522013}]#01211<-(3588877378925522013){'2020-03-31 21:05:11',NULL,'RUNQUEUED','ParentJob','2020-03-31 21:05:16','2020-03-31 21:05:14'}#01211:[3588877378925522013..3588877378925522013]#01213:[{3588877378925522013,'PAUSED'}..{3588877378925522013,'PAUSED'}]#01211<-(7817508156870089629){'2020-03-31 21:05:11',NULL,'FINISHED','ChildJob','2020-03-31 21:05:11','2020-03-31 21:05:14'}#01211:[7817508156870089629..7817508156870089629]#01213:[{3588877378925522013,'FINISHED'}..{3588877378925522013,'FINISHED'}]#01213:[{3588877378925522013,'CANCELLED'}..{3588877378925522013,'CANCELLED'}]#01211:[3588877378925522013..3588877378925522013]
```

### CODE REVIEW

**Comment by tylerkaraszewski** [comment]
I think this might log sensitive info that we're not allowed to log.

**Comment by coleaeason** [comment]
can you change this to log to a separate file that isn't syslog like we do for the other query logging stuff we have in sqlite.cpp? then it doesn't matter what it logs. All truly sensitive values are encrypted before they are inserted, so really it's just going to log a bunch of emails.

File: sqlitecluster/SQLite.cpp [hunk 1]
```diff
@@ -760,6 +779,12 @@ int SQLite::commit() {
     // If there were conflicting commits, will return SQLITE_BUSY_SNAPSHOT
     SASSERT(result == SQLITE_OK || result == SQLITE_BUSY_SNAPSHOT);
     if (result == SQLITE_OK) {
+        if (_currentTransactionAttemptCount != -1) {
+            string logLine = SWHEREAMI + "[row-level-locking] transaction attempt:" +
```
Comment (coleaeason) [inline sqlitecluster/SQLite.cpp:783]: this is confusing to me. this says "transaction attempt" which to me implies it's the attempt number for a given transaction, but the number increases forever until the binary is restarted. so it's not actually the attempt for this transaction.

Comment (tylerkaraszewski) [reply sqlitecluster/SQLite.cpp:783]: It's the attempt for all transactions, like `commitCount`, but including failed commits.

File: BedrockServer.cpp [hunk 2]
```diff
@@ -701,7 +701,7 @@ void BedrockServer::worker(const SData& args,
     // Worker 0 is the "blockingCommit" thread.
     SInitialize(threadId ? "worker" + to_string(threadId) : "blockingCommit");
     int64_t mmapSizeGB = args.isSet("-mmapSizeGB") ? stoll(args["-mmapSizeGB"]) : 0;
-    SQLite db(args["-db"], args.calc("-cacheSize"), false, args.calc("-maxJournalSize"), threadId, threadCount - 1, args["-synchronous"], mmapSizeGB);
+    SQLite db(args["-db"], args.calc("-cacheSize"), false, args.calc("-maxJournalSize"), threadId, threadCount - 1, args["-synchronous"], mmapSizeGB, args.test("-pageLogging"));
```
Comment (coleaeason) [inline BedrockServer.cpp:704]: don't we want to track for sync thread too?

Comment (tylerkaraszewski) [reply BedrockServer.cpp:704]: Yes. Will update.

**Comment by coleaeason** [comment]
Bump?

**Comment by tylerkaraszewski** [comment]
@coleaeason updated.

**Pull request merged**